cmd/snap/cmd_interfaces.go

```diff
@@ -59,34 +59,61 @@ func init() {
 }
 
 func (x *cmdInterfaces) Execute(args []string) error {
-	plugs, err := Client().AllPlugs()
+	conns, err := Client().InterfaceConnections()
 	if err == nil {
 		w := tabwriter.NewWriter(Stdout, 0, 4, 1, ' ', 0)
 		fmt.Fprintln(w, i18n.G("plug\tslot"))
 		defer w.Flush()
-		for _, plug := range plugs {
+		for _, plug := range conns.Plugs {
 			if x.Positionals.Query.Snap != "" && x.Positionals.Query.Snap != plug.Snap {
 				continue
 			}
-			if x.Positionals.Query.Name != "" && x.Positionals.Query.Name != plug.Plug.Name {
+			if x.Positionals.Query.Name != "" && x.Positionals.Query.Name != plug.Name {
 				continue
 			}
 			if x.Interface != "" && plug.Interface != x.Interface {
 				continue
 			}
-			fmt.Fprintf(w, "%s:%s\t", plug.Snap, plug.Plug.Name)
+			// The OS snap (always ubuntu-core) is special and enable abbreviated
+			// display syntax on the plug-side of the connection.
+			if plug.Snap != "ubuntu-core" {
```

Comment:
Nitpick suggestion for coding style: in most cases it's easier for humans to read branching logic on the positive side.

So, first I'd list the "snap _is_ ubuntu-core" one, then the alternative.